src/native/aws_signing.c

```diff
@@ -492,6 +503,74 @@ void JNICALL Java_software_amazon_awssdk_crt_auth_signing_AwsSigner_awsSignerSig
     (*env)->ExceptionClear(env);
 }
 
+JNIEXPORT
+void JNICALL Java_software_amazon_awssdk_crt_auth_signing_AwsSigner_awsSignerSignTrailingHeaders(
+    JNIEnv *env,
+    jclass jni_class,
+    jbyteArray marshalled_headers,
+    jbyteArray java_previous_signature,
+    jobject java_signing_config,
+    jobject java_signing_result_future) {
+
+    (void)jni_class;
+    (void)env;
+
+    struct aws_allocator *allocator = aws_jni_get_allocator();
+    struct s_aws_sign_request_callback_data *callback_data =
+        aws_mem_calloc(allocator, 1, sizeof(struct s_aws_sign_request_callback_data));
+    /* we no longer worry about allocation failures */
+
+    jint jvmresult = (*env)->GetJavaVM(env, &callback_data->jvm);
+    AWS_FATAL_ASSERT(jvmresult == 0);
+
+    callback_data->java_signing_result_future = (*env)->NewGlobalRef(env, java_signing_result_future);
+    AWS_FATAL_ASSERT(callback_data->java_signing_result_future != NULL);
+
+    callback_data->trailing_headers = aws_http_headers_new_from_java_http_headers(env, marshalled_headers);
+    if (callback_data->trailing_headers == NULL) {
+        aws_jni_throw_runtime_exception(env, "Error building headers");
```

Comment:
The `aws_http_headers_new_from_java_http_headers` already throw exceptions on failure, do we need to throw another one here? I am actually not sure about it....

Reply:
good question, @bretambrose do you have thoughts on this?

Reply:
Can probably drop this throw.